library/core/src/ffi/c_str.rs

```diff
@@ -252,7 +250,8 @@ impl CStr {
     #[inline]
     #[must_use]
     #[stable(feature = "rust1", since = "1.0.0")]
-    #[rustc_const_unstable(feature = "const_cstr_methods", issue = "101719")]
+    #[rustc_allow_const_fn_unstable(const_eval_select)]
```

Comment:
Cc @rust-lang/wg-const-eval

Reply:
This is the first time we stabilize a usage of `const_eval_select` that is not just changing assertions. While it is absolutely benign here (basically inlining a naive `strlen` impl for const eval), this means if we'd remove `const_eval_select`, we'd need to either take a perf hit in the runtime code by using the same code or make `strlen` a lang item.

Reply:
then we should "stabilize" `const_eval_select` as a perma-unstable libs function, to affirm that no matter whether it is exposed or not to end users it would still exist in the standard library.

Reply:
How big of a discussion would that be? I.e., should I open a separate PR that does this, or just remove this method from the diff of this PR?

Reply:
I think removing it for now is the easiest. There's an ongoing RFC for stabilizing that intrinsic (without exposing it as public API), so let's just let that RFC handle it